resources/references/adr/2023-09-22-catalog-import-api.md

```diff
@@ -0,0 +1,290 @@
+
+As previously noted, records will be batched by workers according to their level, traversing from the highest level to the lowest. During this process, the generic record structure will be mapped to the expected DAL structure.
+
+Internally, the sync API will be used to perform the actual "inserts".
+
+Calling commit can only be performed once. Any subsequent commits will return an error.
+
+### Getting the status of an import 
+
+`POST /api/import/{{import_id}}/status`
+
+This endpoint returns the status of the import.
+
+Subsequent calls to the status end point will return the current status of the import and errors reported so far.
+
+The status will be reported as one of `started`, `importing`, `cancelled` or `done`. 
+
+When the import is finished, `status` will be reported as `done`, and all the errors will be included and the duration and totals will reflect the total imported.
+
+### Cancelling an import 
+
+`POST /api/import/{{import_id}}/cancel`
+
+This endpoint cancels a non-committed import. It's not possible to cancel an already committed import.
+
+Cancelling an import will delete all records pushed to it. This might be useful in case some erroneous data was pushed to the import.
+
+### Http Schema
+[Onpenapi schema](../../../assets/adr/catalog-import/import-http-schema.yml)
+
+### Resolver Concept
+
+When associating records it will be possible to specify the ID of the association, for example, when you want to assign a tax entity to a product, you can pass the tax ID.
+
+The problem here is that you may not know, or want to keep a record of the referenced entity ID.
+
+We will introduce the concept of Resolvers to solve this problem.
+
+The job of a resolver is to look for references to its entities (in this case Tax) and resolve any of them which are not ID references, to ID references.
+
+In the initial experimental release, we will ship with a minimal set of reference resolvers to prove the concept. We will also explore how we can open it up for extension developers to provide reference resolvers and have them automatically integrated with the API specification.
+
+## Media Uploading
+
+In the first iteration, we will only support uploading media files from an external URL. The image will be downloaded during import.
+
+We will still honor the configuration setting `media.enable_url_upload_feature`. If the config is disabled, the import will skip importing media.
+
+## Error handling
+
+### Whilst adding import records
+
+Whilst adding import instructions via the '/record' endpoints, errors will be directly reported to you regarding resolving. For example, if you try to update an entity which does not exist. Or assign an entity to another that does not exist.
+
+See [Error Response: Resolving Root Entities](assets/catalog-import/examples#error-response-resolving-root-entities) for the request and response when trying to update a product which does not exist.
```

Comment:
I'd also prefer `assets/` folder to be referenced as `./assets/`, or we need to update the [`update-adrs.sh`](https://github.com/shopware/docs/blob/main/.github/scripts/update-adrs.sh#L15) script.